### PR TITLE
UICR-164 Implemented Hyperlinks and get/set url logic

### PR DESCRIPTION
**Pull request by chris-hellen**
## Purpose
Make the Courses and Reserves tabs more accessible by adding hyperlinks.

## Approach
Adapted formatter to create links for the first column result in the courses/reserves search result MCL. Links retain same functionality as clicking row did before the change. Also added local function to retrieve URL and set URL for each item.

## Refs
[UICR-164](https://folio-org.atlassian.net/browse/UICR-164)

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added an appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

### CODE REVIEW

File: src/components/Courses.js [hunk 1]
```diff
@@ -260,6 +273,7 @@ class Courses extends React.Component {
     };
 
     const resultsFormatter = {
+      name: r => <TextLink to={this.getRowURL(r.id)} onClick={() => this.setURL(r.id)}>{r.name}</TextLink>,
```
Comment (MikeTaylor) [inline src/components/Courses.js:276]: Why does this use both `to` _and_ `onClick`?

File: src/components/Courses.js [hunk 2]
```diff
@@ -336,13 +350,14 @@ class Courses extends React.Component {
                         visibleColumns={visibleColumns}
                         columnWidths={columnWidths}
                         columnMapping={columnMapping}
+                        resultsRowClickHandlers={false}
                         formatter={resultsFormatter}
+                        resultsRowFormatter={DefaultMCLRowFormatter}
```
Comment (MikeTaylor) [inline src/components/Courses.js:355]: I'm interested to know why this is necessary: isn't it the default
?

**Review by MikeTaylor — COMMENTED** [review]
Don't forget to add a change-log entry!

File: src/components/Courses.js (hunk 1 above)
Comment (chris-hellen) [reply src/components/Courses.js:276]: I was following the same logic that was used in the PR referenced in the JIRA ticket,[UIREQ-716](https://github.com/folio-org/ui-requests/pull/849/commits/0afdcb82f0d3ae3e0b4a8bac1d3f1b6296c80297). `to` is used to serve the row URL and `onClick` is used to save the URL in a state variable (which I am adding in this next commit, apologies I forgot to add it in the first commit). Is there a more efficient way I could be doing this?

File: src/components/Courses.js (hunk 2 above)
Comment (chris-hellen) [reply src/components/Courses.js:355]: You are correct, it is the default. Deleting, thank you!

**Comment by chris-hellen** [comment]
> Don't forget to add a change-log entry!

Sorry, I am not familiar with adding change-log entries. I am assuming I would add it under 6.0.3?

File: src/components/Courses.js (hunk 1 above)
Comment (MikeTaylor) [reply src/components/Courses.js:276]: I don't understand why you're saving the URL at all. That bit of state doesn't seem to get used (as is reported by GitHub's built-in linter). Am I missing something?

**Comment by MikeTaylor** [comment]
> > Don't forget to add a change-log entry!
> 
> Sorry, I am not familiar with adding change-log entries. I am assuming I would add it under 6.0.3?

You just add a line to the file `CHANGELOG.md` under the heading for the next release (which will be the first one containing this enhancement).

That would be 6.1.0, an in-progress minor release. 6.0.3 has already been released and should be further modified. And there's no need for a new 6.0.4 since this is not an urgent bugfix but an enhancement.

File: src/components/Courses.js (hunk 1 above)
Comment (chris-hellen) [reply src/components/Courses.js]: No, you are correct. I was mistaken, the setURL was not necessary after looking back at the functionality. Newest commit should fix all the current issues. Thank you!

**Review by MikeTaylor — APPROVED** [review]
Much better!

(Do beware of cargo-cult programming: using the changes from another commit without fully understanding them.)

**Comment by chris-hellen** [comment]
Than you for your help, I appreciate it! Last question, how do I go about getting the checks to pass so that I am able to merge?

**Comment by MikeTaylor** [comment]
> Than you for your help, I appreciate it! Last question, how do I go about getting the checks to pass so that I am able to merge?

I'm afraid I don't know, I've not seen that particular error before. Perhaps @zburke or @skomorokh can help? (I take a quick moment of sadness that in the past I would also have tagged @mkuklis and @doytch :-( )

**Comment by zburke** [comment]
@chris-hellen, @MikeTaylor, TLDR some GithubActions require organization secrets and won't run from forked repositories. I pulled the branch from the fork and re-pushed it under folio-org and opened a PR, #286. Once the GAs finish running there you should be able to merge the original PR and close the copy.**

This solution is outlined in our [developer documentation](https://dev.folio.org/guidelines/github-actions-frontend/#forked-repositories). It works because the commit hashes on my branch will match those on the original, and GH is smart enough to notice this, so it marks the corresponding tasks as passing on both PRs. The failure here is oddly silent; I feel like we used to get more info when Sonar failed, but it's been a while and maybe, like Mike, I'm just wistfully remembering the past. 

** A secondary problem on my branch occurred at the next step: we got rate-limited by GitHub, preventing the (successful) test results from being published. See https://github.com/folio-org/.github/blob/master/README-UI.md for instructions on how to migrate to the new GAs. Tag me on the PR if you pursue this; devops have locked down the protected branch settings and you'll need somebody with access to flip from the old required tasks to the new ones once the PR is open.

File: src/components/Reserves.js [hunk 3]
```diff
@@ -96,6 +100,15 @@ class Reserves extends React.Component {
     this.props.history.push(`/cr/reserves/${row.courseListingId}/0/${row.id}/${row.itemId}/edit${this.props.location.search}`);
   }
 
+  getRowURL(row) {
```
Comment (zburke) [inline src/components/Reserves.js:103]: pedantic overthinking: This function configures a link to a reserve. Practically, of course, it doesn't matter whether you use names like `getRowURL(row)` or `getReserveURL(reserve)`, but the way you think about it is important. this hierarchy goes only one way; an MCL container knows about its rows and a row about its record, but the reverse is not true: a record doesn't know if it's being displayed in a row or a list or a headline. 

You don't need to change anything here, it's not worth the effort. But as you build new things, being deliberate about this kind of organization can have a clarifying effect on the rest of your code. When you are particular about the properties of a reserve, of a row containing a reserve, of a table containing a list of rows, then it becomes very obvious where to add a new function when you are tinkering, or where to look for the source of trouble when you are debugging.

**Review by zburke — APPROVED** [review]

**Review by zburke — COMMENTED** [review]
SonarCloud would like to see a test to cover this new function. Regrettably, there are zero tests for this component in the repository already 😢 and it feels unfair to put that whole burden on you for this small change. 

That said, a small test that inspects the page attributes based on the presence/absence of the storage permissions here should not take much effort: 
```
expect(screen.getByRole('link', { name: copiedItem.title })).toHaveAttribute('href', 'some/reserve/properties');
expect(screen.queryByRole('link', { name: copiedItem.title })).toBeNull();
```

**Pull request merged**